### ISSUE TITLE
Is it possible to concurrently use more than one network configuration?

### ISSUE THREAD

**Issue by edunuzzi** [stellar/java-stellar-sdk#155]
Hi, i am trying to figure out if i could use multiple network configurations, concurrently. Our application works with both a private network and the Stellar public network. It would be really great if we could switch dynamically between those different configurations on demand, in a way that works with concurrent requests. Any ideas of how something like this could work in the existing implementation?

**Comment by jillesvangurp** [comment]
Not currently, this seems to be managed with a global, static variable via

```
 Network.use(Network(networkPassphrase))
// ore
Network.useTestNetwork()
Network.usePublicNetwork()
```

IMHO this should be a constructor argument to `Server`. Then you could simply create a new instance with a different network.

I'd recommend getting rid of that; BTW this looks a bit hacky:

```
    public byte[] getNetworkId() {
        return Util.hash(current.getNetworkPassphrase().getBytes(Charset.forName("UTF-8")));
    }
```

So that uses a static method to get the static instance and then calls an instance method on that, that again uses the static field to get the network passphrase.  It looks like this is only used deep inside the Transaction class. So, should be relatively easy to fix. 

@bartekn I just created another pull request #182 that adds a new constructor to deal with some okhttpclient issues. Adding the network as a parameter there as well seems like a sensible solution. If you want, I can take care of that in that pr; or we can do a new one after you approve that one. Let me know what you want done.

**Comment by edunuzzi** [comment]
> IMHO this should be a constructor argument to `Server`. Then you could simply create a new instance with a different network.

That would be a great solution in my opinion too. Also, that is the way that SDKs in other languages deal with that. 

Unless we have another suggestion, i would be willing to create a pull request with the required changes.

**Comment by bnogalm** [comment]
I vote for the Server argument, after all, you set the URI to the server in the constructor and you only need the network id to sign transactions.
"Transaction Builder" should be an object created by a Server's method.

Removing the static Network id will also help multithreading.

**Comment by overcat** [comment]
I submitted a PR related to this, does this look okay? This modification provides forward compatibility.

I have not written test code yet.

Example
```
        Transaction transaction = new Transaction.Builder(sourceAccount)
                .addOperation(new ManageDataOperation.Builder("hello", "world".getBytes()).build())
                .setTimeout(Transaction.Builder.TIMEOUT_INFINITE)
                .setNetwork(Network.testNetwork())  # use test network
                .setOperationFee(100)
                .build();
        transaction.sign(source);
```

**Comment by overcat** [comment]
Implemented in #224
